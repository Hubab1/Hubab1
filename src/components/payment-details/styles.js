import { css } from 'emotion';

export const cardHeader = css`
    align-self: start;
    flex: 1;
    padding-left: 0;
    font-weight: 400;
    display: block;
    font-size: 14px;
`;

export const headerContainer = css`
    display: flex;
    flex: 1;
    align-items: center;
    text-transform: none;
    padding-bottom: 15px;
`;

export const imageWrapper = css`
    line-height: 67px;
    width: 42px;
    text-align: left;
    margin: 0;
    padding: 0;
    margin-right: 10px;
`;

export const itemsContainer = css`
    margin-bottom: 0;
    border-bottom: 1px solid #eeeeee;

    .MuiExpansionPanelSummary-root {
        padding: 0;
    }

    .MuiExpansionPanelSummary-root.Mui-expanded {
        min-height: 20px;
        height: 45px;
        margin: 0;
    }

    :nth-child(2) {
        border-top: 1px solid #eeeeee;
    }
`;

export const itemRow = css`
    padding: 10px 0;
    text-align: left;
    border-top: 1px solid #eeeeee;
    font-size: 15px;
`;

export const paidText = css`
    position: absolute;
    right: 28px;
    top: 13px;
`;

export const paymentDetailRow = css`
    display: flex;
    justify-content: space-between;
    padding-bottom: 9px;
    font-size: 14px;

    div:first-child {
        padding-right: 50px;
    }
`;

export const paymentDetailTotal = css`
    display: flex;
    justify-content: space-between;
    background-color: rgba(103, 193, 139, 0.1);
    padding: 10px 15px 10px 16px;
    width: 458px;
    margin-left: -15px;
    font-size: 15px;

    div:first-child {
        font-weight: bold;
    }
`;

export const title = css`
    font-weight: bold;
`;

export const rentalOptionName = css`
    padding-right: 20px;
`;

export const styles = {
    cardHeader,
    headerContainer,
    itemsContainer,
    itemRow,
    imageWrapper,
    paidText,
    paymentDetailRow,
    paymentDetailTotal,
    title,
    rentalOptionName,
};

export const viewPaymentTerms = css`
    color: #2b44ff;
    cursor: pointer;
    text-decoration: underline;
    padding-left: 5px;
`;

export const termsDiv = css`
    margin-bottom: 20px;
    margin-top: 44px;
    font-size: 14px;
`;
