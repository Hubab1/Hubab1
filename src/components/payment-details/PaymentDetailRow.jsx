import PropTypes from 'prop-types';
import React from 'react';
import { styles } from './styles';
import { getPaymentItemName } from 'utils/misc';

export const TYPE_CONCESSION = 'concession';

export function PaymentDetailRows({ paymentObject, paymentType }) {
    const rows = paymentObject.items.map((item) => (
        <PaymentDetailRow
            key={`${item.name}-${paymentType}`}
            className={styles.paymentDetailRow}
            name={item.name}
            paymentTotal={item.amount}
            quantity={item.quantity}
            price={item.price}
            included={item.included}
            perDay={item.per_day}
            prorated={item.prorated}
            days={item.days}
            exempted={item.exempted}
            type={item.type}
        />
    ));

    rows.push(
        <PaymentDetailRow
            key={`total${paymentType}`}
            name="Total"
            paymentTotal={paymentObject.total}
            className={styles.paymentDetailTotal}
        />
    );

    return rows;
}

PaymentDetailRows.propTypes = {
    paymentObject: PropTypes.object.isRequired,
    paymentType: PropTypes.string.isRequired,
};

export const PaymentDetailRow = ({
    name,
    paymentTotal,
    quantity = 1,
    price,
    included = 0,
    className,
    prorated,
    days,
    perDay,
    exempted,
    type,
}) => {
    if (!quantity) return null;

    return (
        <div className={className}>
            <div>
                {getRentalOptionFeeDisplayName(name, price, quantity, included, prorated, days, perDay, exempted)}
            </div>
            <div>
                {type === TYPE_CONCESSION && `-`}${paymentTotal}
            </div>
        </div>
    );
};

PaymentDetailRow.propTypes = {
    className: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    paymentTotal: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
    quantity: PropTypes.number,
    price: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    included: PropTypes.number,
    prorated: PropTypes.bool,
    days: PropTypes.number,
    perDay: PropTypes.string,
    exempted: PropTypes.number,
    type: PropTypes.string,
};

// This should match the logic in chuck
export function getRentalOptionFeeDisplayName(
    name,
    price,
    quantity = 1,
    included = 0,
    prorated,
    days,
    perDay,
    exempted = 0
) {
    if (quantity < 1) return null;

    const feesBreakdown = [];
    const num_included = quantity > included ? included : quantity;
    if (num_included > 0) {
        feesBreakdown.push(`${num_included} incl.`);
    }

    if (exempted) {
        feesBreakdown.push(`${exempted} service animal${exempted > 1 ? 's' : ''} incl.`);
    }

    const uncharged_quantity = included + exempted;
    const num_charged = quantity > uncharged_quantity ? quantity - uncharged_quantity : 0;
    if ((num_charged > 0 || exempted > 0) && quantity > 1 && !prorated) {
        feesBreakdown.push(`${num_charged} x $${price}`);
    }

    let displayName = `${getPaymentItemName(name)}`;
    const daysPlural = days > 1 ? 's' : '';

    if (prorated && num_charged > 0 && quantity > 0) {
        displayName += `: ${days} day${daysPlural} at $${perDay}/day`;
    }

    if (feesBreakdown.length > 0) {
        displayName += ` (${feesBreakdown.join(', ')})`;
    }

    return displayName;
}
