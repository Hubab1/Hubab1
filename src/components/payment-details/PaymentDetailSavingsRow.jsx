import { styles } from 'components/payment-details/styles';
import PropTypes from 'prop-types';
import React from 'react';

export function PaymentDetailSavingsRows({ paymentObject, paymentType }) {
    const rows = paymentObject.items.map((item) => (
        <PaymentDetailSavingsRow
            key={`${item.name}-${paymentType}`}
            name={item.name}
            paymentTotal={item.amount}
            pricePerMonth={item.price}
            months={item.months}
            className={styles.paymentDetailRow}
        />
    ));
    rows.push(
        <PaymentDetailSavingsRow
            key={`total${paymentType}`}
            name={'Total'}
            paymentTotal={paymentObject.total}
            className={styles.paymentDetailTotal}
        />
    );
    return rows;
}

PaymentDetailSavingsRows.propTypes = {
    paymentObject: PropTypes.object.isRequired,
    paymentType: PropTypes.string.isRequired,
};

export const PaymentDetailSavingsRow = ({ paymentTotal, name, months, pricePerMonth, className }) => {
    if (name.toLowerCase() === 'rentable item concession') {
        name = 'Parking, Storage, Other Monthly Charge Concession';
    }

    return (
        <div className={className}>
            <div>
                {name}
                {months && `: ${months} months at $${pricePerMonth}/month`}
            </div>
            <div>${paymentTotal}</div>
        </div>
    );
};

PaymentDetailSavingsRow.propTypes = {
    name: PropTypes.string.isRequired,
    paymentTotal: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
    months: PropTypes.number,
    pricePerMonth: PropTypes.string,
};
