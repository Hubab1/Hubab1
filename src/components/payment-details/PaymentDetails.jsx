import React from 'react';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { ROUTES } from 'app/constants';
import { Card, H1, SpacedH3 } from 'assets/styles';
import captureRoute from 'app/captureRoute';
import lightbulb from 'assets/images/lightbulb.png';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';

import { makeStyles } from '@material-ui/core/styles';
import { PaymentDetailRows } from 'components/payment-details/PaymentDetailRow';
import { styles } from './styles';
import { format, parseISO } from 'date-fns';

const useStyles = makeStyles(() => ({
    root: {
        display: 'block',
        padding: 0,
    },
}));

export const PaymentCard = styled(Card)`
    padding: 15px;
`;

export const PaymentDetails = ({ profile, configuration }) => {
    if (!profile || !configuration || !profile.lease_start_date) return null;

    const leaseStartDate = format(parseISO(profile.lease_start_date), 'MM/dd/yyyy');

    return (
        <>
            <H1>Payment Details</H1>
            <SpacedH3>Hereâ€™s the breakdown for unit {profile.unit.unit_number}</SpacedH3>
            <PaymentCard>
                <div className={styles.headerContainer}>
                    <div className={styles.imageWrapper}>
                        <img alt="coin" src={lightbulb} />
                    </div>
                    <div className={styles.cardHeader}>
                        <div className={styles.title}>Payment breakdown</div>
                        <div>Move in date: {leaseStartDate}</div>
                        <div>Lease term: {profile.lease_term} months</div>
                    </div>
                </div>
                <PaymentItemsExpansionPanel
                    amount={profile.fees_breakdown.application_fees.total}
                    label={'due at application'}
                    defaultExpanded={false}
                >
                    <PaymentDetailRows
                        paymentObject={profile.fees_breakdown.application_fees}
                        paymentType="application"
                    />
                </PaymentItemsExpansionPanel>
                <PaymentItemsExpansionPanel
                    amount={profile.fees_breakdown.move_in_fees.total}
                    label={'due at move in'}
                    defaultExpanded={false}
                >
                    <PaymentDetailRows paymentObject={profile.fees_breakdown.move_in_fees} paymentType="move_in" />
                </PaymentItemsExpansionPanel>
                <PaymentItemsExpansionPanel
                    amount={profile.fees_breakdown.monthly_fees.total}
                    label={'monthly rent'}
                    defaultExpanded={false}
                >
                    <PaymentDetailRows paymentObject={profile.fees_breakdown.monthly_fees} paymentType="monthly_fees" />
                </PaymentItemsExpansionPanel>
            </PaymentCard>
        </>
    );
};

PaymentDetails.propTypes = {
    profile: PropTypes.object.isRequired,
    configuration: PropTypes.object.isRequired,
};

function PaymentItemsExpansionPanel({ children, defaultExpanded, label, amount }) {
    const classes = useStyles();

    return (
        <div className={styles.itemsContainer}>
            <ExpansionPanel elevation={0} defaultExpanded={defaultExpanded}>
                <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
                    <b>${amount}</b>&nbsp;{label}
                </ExpansionPanelSummary>
                <ExpansionPanelDetails classes={{ root: classes.root }}>
                    {React.Children.map(children, (child) => (
                        <div className={styles.itemRow}>{child}</div>
                    ))}
                </ExpansionPanelDetails>
            </ExpansionPanel>
        </div>
    );
}

PaymentItemsExpansionPanel.propTypes = {
    children: PropTypes.object,
    defaultExpanded: PropTypes.bool,
    label: PropTypes.string.isRequired,
    amount: PropTypes.string.isRequired,
};

const mapStateToProps = (state) => ({
    profile: state.renterProfile,
    configuration: state.configuration,
});

export default connect(mapStateToProps)(captureRoute(PaymentDetails, ROUTES.PAYMENT_DETAILS));
