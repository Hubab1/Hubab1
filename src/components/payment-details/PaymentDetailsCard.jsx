import { styles } from 'components/payment-details/styles';
import styled from '@emotion/styled';
import lightbulb from 'assets/images/lightbulb.png';
import { PaymentDetailRows } from 'components/payment-details/PaymentDetailRow';
import React from 'react';
import { format, parseISO } from 'date-fns';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import PaidText from 'components/common/PaidText';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import { Card } from 'assets/styles';

const useStyles = makeStyles(() => ({
    root: {
        display: 'block',
        padding: 0,
    },
}));

export const PaymentCard = styled(Card)`
    padding: 15px;
`;

export const PaymentDetailsCard = ({ profile, payables = [] }) => {
    const leaseStartDate = format(parseISO(profile.lease_start_date), 'MM/dd/yyyy');
    const allPaymentsPaid = !payables.find((payment) => !payment.paid);

    return (
        <PaymentCard>
            <div className={styles.headerContainer}>
                <div className={styles.imageWrapper}>
                    <img alt="coin" src={lightbulb} />
                </div>
                <div className={styles.cardHeader}>
                    <div className={styles.title}>Payment breakdown</div>
                    <div>Move in date: {leaseStartDate}</div>
                    <div>Lease term: {profile.lease_term} months</div>
                </div>
            </div>
            <PaymentItemsExpansionPanel
                amount={profile.fees_breakdown.application_fees.total}
                label={'due at application'}
                defaultExpanded={false}
                isPaid={allPaymentsPaid}
            >
                <PaymentDetailRows paymentObject={profile.fees_breakdown.application_fees} paymentType="application" />
            </PaymentItemsExpansionPanel>
            <PaymentItemsExpansionPanel
                amount={profile.fees_breakdown.move_in_fees_v2.total}
                label={'due at move in'}
                defaultExpanded={false}
            >
                <PaymentDetailRows paymentObject={profile.fees_breakdown.move_in_fees_v2} paymentType="move_in" />
            </PaymentItemsExpansionPanel>
            <PaymentItemsExpansionPanel
                amount={profile.fees_breakdown.monthly_fees.total}
                label={'monthly rent'}
                defaultExpanded={false}
            >
                <PaymentDetailRows paymentObject={profile.fees_breakdown.monthly_fees} paymentType="monthly_fees" />
            </PaymentItemsExpansionPanel>
        </PaymentCard>
    );
};

PaymentDetailsCard.propTypes = {
    profile: PropTypes.object.isRequired,
    payables: PropTypes.array,
};

export function PaymentItemsExpansionPanel({ children, defaultExpanded, label, amount, isPaid }) {
    const classes = useStyles();

    return (
        <div className={styles.itemsContainer}>
            <ExpansionPanel elevation={0} defaultExpanded={defaultExpanded}>
                <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
                    <div className={styles.itemsContainerLabel}>
                        <b>${amount}</b>&nbsp;{label}
                    </div>
                    {isPaid && (
                        <div className={styles.paidText}>
                            <PaidText />
                        </div>
                    )}
                </ExpansionPanelSummary>
                <ExpansionPanelDetails classes={{ root: classes.root }}>
                    {React.Children.map(children, (child) => (
                        <div className={styles.itemRow}>{child}</div>
                    ))}
                </ExpansionPanelDetails>
            </ExpansionPanel>
        </div>
    );
}

PaymentItemsExpansionPanel.propTypes = {
    children: PropTypes.object,
    defaultExpanded: PropTypes.bool,
    label: PropTypes.string.isRequired,
    amount: PropTypes.string.isRequired,
    isPaid: PropTypes.bool,
};
