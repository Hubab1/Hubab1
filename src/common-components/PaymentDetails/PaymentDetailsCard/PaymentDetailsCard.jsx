import React from 'react';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import { makeStyles, ExpansionPanel, ExpansionPanelSummary, ExpansionPanelDetails } from '@material-ui/core';
import { ExpandMore as ExpandMoreIcon } from '@material-ui/icons';

import { format, parseISO } from 'date-fns';

import PaidText from 'common-components/PaidText/PaidText';
import { PaymentDetailRows } from '../PaymentDetailsRow/PaymentDetailRow';
import { PaymentDetailSavingsRows } from '../PaymentDetailSavingsRows/PaymentDetailSavingsRow';
import { Card } from 'assets/styles';
import { styles } from '../styles';
import lightbulb from 'assets/images/lightbulb.png';

const useStyles = makeStyles(() => ({
    root: {
        display: 'block',
        padding: 0,
    },
}));

export const PaymentCard = styled(Card)`
    padding: 15px;
`;

export const PaymentDetailsCard = ({ profile, applicationFees }) => {
    const leaseStartDate = format(parseISO(profile.lease_start_date), 'MM/dd/yyyy');

    return (
        <PaymentCard>
            <div className={styles.headerContainer}>
                <div className={styles.imageWrapper}>
                    <img alt="coin" src={lightbulb} />
                </div>
                <div className={styles.cardHeader}>
                    <div className={styles.title}>Payment breakdown</div>
                    <div>Move in date: {leaseStartDate}</div>
                    <div>Lease term: {profile.lease_term} months</div>
                </div>
            </div>
            <PaymentItemsExpansionPanel
                amount={applicationFees.total}
                label={'due at application'}
                defaultExpanded={false}
                isPaid={applicationFees.allPaid}
            >
                <PaymentDetailRows paymentObject={applicationFees} paymentType="application" />
            </PaymentItemsExpansionPanel>
            <PaymentItemsExpansionPanel
                amount={profile.fees_breakdown.move_in_fees.total}
                label={'due at move in'}
                defaultExpanded={false}
            >
                <PaymentDetailRows paymentObject={profile.fees_breakdown.move_in_fees} paymentType="move_in" />
            </PaymentItemsExpansionPanel>
            <PaymentItemsExpansionPanel
                amount={profile.fees_breakdown.monthly_fees.total}
                label={'monthly rent'}
                defaultExpanded={false}
            >
                <PaymentDetailRows paymentObject={profile.fees_breakdown.monthly_fees} paymentType="monthly_fees" />
            </PaymentItemsExpansionPanel>
            {profile.fees_breakdown.savings?.items.length > 0 && (
                <PaymentItemsExpansionPanel
                    amount={profile.fees_breakdown.savings.total}
                    label={'savings'}
                    defaultExpanded={false}
                >
                    <PaymentDetailSavingsRows paymentObject={profile.fees_breakdown.savings} paymentType="savings" />
                </PaymentItemsExpansionPanel>
            )}
        </PaymentCard>
    );
};

PaymentDetailsCard.propTypes = {
    profile: PropTypes.object.isRequired,
    payables: PropTypes.array,
};

export function PaymentItemsExpansionPanel({ children, defaultExpanded, label, amount, isPaid }) {
    const classes = useStyles();

    return (
        <div className={styles.itemsContainer}>
            <ExpansionPanel elevation={0} defaultExpanded={defaultExpanded}>
                <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
                    <div className={styles.itemsContainerLabel}>
                        <b>${amount}</b>&nbsp;{label}
                    </div>
                    {isPaid && (
                        <div className={styles.paidText}>
                            <PaidText />
                        </div>
                    )}
                </ExpansionPanelSummary>
                <ExpansionPanelDetails classes={{ root: classes.root }}>
                    {React.Children.map(children, (child) => (
                        <div className={styles.itemRow}>{child}</div>
                    ))}
                </ExpansionPanelDetails>
            </ExpansionPanel>
        </div>
    );
}

PaymentItemsExpansionPanel.propTypes = {
    children: PropTypes.object,
    defaultExpanded: PropTypes.bool,
    label: PropTypes.string.isRequired,
    amount: PropTypes.string.isRequired,
    isPaid: PropTypes.bool,
    applicationFees: PropTypes.object,
};
