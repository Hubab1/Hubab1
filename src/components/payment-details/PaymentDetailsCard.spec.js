import { shallow } from 'enzyme';
import React from 'react';
import { PaymentDetailsCard, PaymentItemsExpansionPanel } from 'components/payment-details/PaymentDetailsCard';

describe('PaymentDetails', () => {
    let defaultProps;

    beforeEach(() => {
        defaultProps = {
            profile: {
                lease_start_date: '2020-12-15',
                unit: {
                    unit_number: 1,
                },
                lease_term: 12,
                fees_breakdown: {
                    application_fees: {
                        total: '620',
                        items: [],
                    },
                    move_in_fees_v2: {
                        total: '0',
                        items: [],
                    },
                    monthly_fees: {
                        total: '1234',
                        items: [],
                    },
                },
            },
            payables: [{ paid: true }, { paid: true }, { paid: true }],
        };
    });

    it('matches snapshot without savings', () => {
        const wrapper = shallow(<PaymentDetailsCard {...defaultProps} />);
        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.find(PaymentItemsExpansionPanel)).toHaveLength(3);
    });

    it('matches snapshot with savings', () => {
        const props = defaultProps;
        props.profile.fees_breakdown.savings = {
            items: [
                {
                    name: 'Rental fee waiver',
                    amount: '2,500',
                    price: '250',
                    months: '10',
                },
                {
                    name: 'Security deposit waiver',
                    amount: '250',
                    price: '250',
                },
            ],
            total: '2,500',
        };
        const wrapper = shallow(<PaymentDetailsCard {...props} />);
        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.find(PaymentItemsExpansionPanel)).toHaveLength(4);
    });
});
