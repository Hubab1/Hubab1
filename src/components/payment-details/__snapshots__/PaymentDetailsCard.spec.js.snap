// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentDetails matches snapshot with savings 1`] = `
<ForwardRef(render)>
  <div
    className="css-1umq6o3"
  >
    <div
      className="css-vutzwz"
    >
      <img
        alt="coin"
        src="lightbulb.png"
      />
    </div>
    <div
      className="css-1s5wtrg"
    >
      <div
        className="css-eizsos"
      >
        Payment breakdown
      </div>
      <div>
        Move in date: 
        12/15/2020
      </div>
      <div>
        Lease term: 
        12
         months
      </div>
    </div>
  </div>
  <PaymentItemsExpansionPanel
    amount="620"
    defaultExpanded={false}
    isPaid={true}
    label="due at application"
  >
    <PaymentDetailRows
      paymentObject={
        Object {
          "items": Array [],
          "total": "620",
        }
      }
      paymentType="application"
    />
  </PaymentItemsExpansionPanel>
  <PaymentItemsExpansionPanel
    amount="0"
    defaultExpanded={false}
    label="due at move in"
  >
    <PaymentDetailRows
      paymentObject={
        Object {
          "items": Array [],
          "total": "0",
        }
      }
      paymentType="move_in"
    />
  </PaymentItemsExpansionPanel>
  <PaymentItemsExpansionPanel
    amount="1234"
    defaultExpanded={false}
    label="monthly rent"
  >
    <PaymentDetailRows
      paymentObject={
        Object {
          "items": Array [],
          "total": "1234",
        }
      }
      paymentType="monthly_fees"
    />
  </PaymentItemsExpansionPanel>
  <PaymentItemsExpansionPanel
    amount="2,500"
    defaultExpanded={false}
    label="savings"
  >
    <PaymentDetailSavingsRows
      paymentObject={
        Object {
          "items": Array [
            Object {
              "amount": "2,500",
              "months": "10",
              "name": "Rental fee waiver",
              "price": "250",
            },
            Object {
              "amount": "250",
              "name": "Security deposit waiver",
              "price": "250",
            },
          ],
          "total": "2,500",
        }
      }
      paymentType="savings"
    />
  </PaymentItemsExpansionPanel>
</ForwardRef(render)>
`;

exports[`PaymentDetails matches snapshot without savings 1`] = `
<ForwardRef(render)>
  <div
    className="css-1umq6o3"
  >
    <div
      className="css-vutzwz"
    >
      <img
        alt="coin"
        src="lightbulb.png"
      />
    </div>
    <div
      className="css-1s5wtrg"
    >
      <div
        className="css-eizsos"
      >
        Payment breakdown
      </div>
      <div>
        Move in date: 
        12/15/2020
      </div>
      <div>
        Lease term: 
        12
         months
      </div>
    </div>
  </div>
  <PaymentItemsExpansionPanel
    amount="620"
    defaultExpanded={false}
    isPaid={true}
    label="due at application"
  >
    <PaymentDetailRows
      paymentObject={
        Object {
          "items": Array [],
          "total": "620",
        }
      }
      paymentType="application"
    />
  </PaymentItemsExpansionPanel>
  <PaymentItemsExpansionPanel
    amount="0"
    defaultExpanded={false}
    label="due at move in"
  >
    <PaymentDetailRows
      paymentObject={
        Object {
          "items": Array [],
          "total": "0",
        }
      }
      paymentType="move_in"
    />
  </PaymentItemsExpansionPanel>
  <PaymentItemsExpansionPanel
    amount="1234"
    defaultExpanded={false}
    label="monthly rent"
  >
    <PaymentDetailRows
      paymentObject={
        Object {
          "items": Array [],
          "total": "1234",
        }
      }
      paymentType="monthly_fees"
    />
  </PaymentItemsExpansionPanel>
</ForwardRef(render)>
`;
